src/commands/check.rs

```diff
@@ -101,180 +87,132 @@ pub struct LintArgs {
-        let lint_rules: HashSet<String> =
-            lint::rules().iter().map(|r| r.id().to_string()).collect();
-        let all_rules: HashSet<_> = analysis_rules.union(&lint_rules).collect();
-
-        let unknown_exceptions: Vec<&str> = exceptions
-            .iter()
-            .filter(|rule| !all_rules.contains(rule))
-            .map(|rule| rule.as_str())
-            .collect();
-
-        if !unknown_exceptions.is_empty() {
-            let (config, writer) =
-                get_display_config(args.common.report_mode, args.common.no_color);
-            let mut w_lock = writer.lock();
-            let files: SimpleFiles<String, String> = SimpleFiles::new();
-
-            let message = format!(
-                "ignoring unknown rule{s} provided via --except: '{rules}'",
-                s = if unknown_exceptions.len() == 1 {
-                    ""
-                } else {
-                    "s"
-                },
-                rules = unknown_exceptions
-                    .iter()
-                    .map(|r| format!("`{r}`"))
-                    .collect::<Vec<_>>()
-                    .join(", ")
-            );
-
-            // TODO: revisit case-insensitive rule handling when wdl crates are updated
-            let warning = codespan_reporting::diagnostic::Diagnostic::warning()
-                .with_message(message)
-                .with_notes(vec![
-                    "run `sprocket explain --help` to see available rules".to_string(),
-                ]);
-
-            term::emit(&mut w_lock, config, &files, &warning)
-                .expect("failed to emit unknown rule warning");
-        }
+    let source_uris = args
+        .common
+        .sources
+        .iter()
+        .flat_map(|s| s.as_url())
+        .cloned()
+        .collect::<HashSet<_>>();
+
+    let results = Analysis::default()
+        .extend_sources(args.common.sources)
+        .extend_exceptions(args.common.except)
+        .lint(args.lint)
+        .shellcheck(args.common.shellcheck)
+        .run()
+        .await?;
+
+    #[derive(Default)]
+    struct Counts {
+        /// The number of errors encountered.
+        pub errors: usize,
+        /// The number of warnings encountered.
+        pub warnings: usize,
+        /// The number of notes encountered.
+        pub notes: usize,
     }
 
-    let results = analyze(file, exceptions, lint, shellcheck).await?;
+    let mut counts = Counts::default();
 
-    let cwd = std::env::current_dir().ok();
-    let mut error_count = 0;
-    let mut warning_count = 0;
-    let mut note_count = 0;
-    for result in &results {
-        let mut suppress = false;
+    for result in results {
+        let uri = &**result.document().uri();
 
-        // Attempt to strip the CWD from the result path
-        let uri = result.document().uri();
-        if args.common.single_document && !uri.as_str().contains(file) {
+        if args.common.explicitly_scoped && !source_uris.contains(uri) {
```

Comment:
I think this will "hide" errors from imported documents, which we don't want to do. `check` should always show all `error` level diagnostics regardless of `explicitly_scoped`

Reply:
That sounds counter-intuitive to me. If I'm providing a flag called `--suppress-imports`, I'm counting on only seeing diagnostics for the documents I'm immediately providing on the command line, effectively ignoring everything outside that universe. If we want to have the behavior you specify here, we should call the flag something different (`--supress-import-notes` and/or `--suppress-import-warnings`).

Personally, I don't really understand why we have the flag in the first place, as this seems like a really niche area. Perhaps we should just remove the flag?

Reply:
I personally use it when I'm writing a workflow that imports a bunch of stuff. I don't want to see diagnostics from the files I'm not currently working on, so I pass `--single-document` to only show me lints from the workflow I'm writing.

Similarly, it's used in `sprocket-action` to more explicitly group diagnostics from the same file/filter out specific files the user doesn't want to see diagnostics from.

The reason to still show errors is that an error means that a `run` command is going to fail due to a bad import. I would be confused if `sprocket check` passes cleanly and then `sprocket run` failed.

Reply:
Alright, I've updated this to now work as you outline.